### PR TITLE
Global viewport histogram improvements + examples

### PR DESCRIPTION
**Pull request by elenatorro**
The purpose of this PR is to add a better example for the Global vs Viewport Histogram and fix related issues we found while using these expressions:

- [x] Reset the histogram when binding the metadata in the GlobalHistogram
- [x] Add "options" parameter to getJoinedValues method (including explanation in the method reference) https://github.com/CartoDB/carto-vl/issues/1267
- [x] Do not display an error when using `getLegendData` method while the viz is blending https://github.com/CartoDB/carto-vl/issues/1271 -> solved by adding a common `getLegendData` method to the base expression (https://github.com/CartoDB/carto-vl/issues/1271#issuecomment-466347103)
- [x] _getLegendData private methods => public
- [x] Fix numeric global histogram applying feature count ratio (@rjimenezda ✨)

### CODE REVIEW

**Comment by elenatorro** [comment]
We've talked about adding an explanation of how the `samples` value in the metadata is calculated in order to add this information to the `globalHistogram` reference (thanks @IagoLast for the suggestion!)

I think @jgoizueta can help us with this :)

**Comment by makella** [comment]
@elenatorro awesome! I will make a few small modifications but first, I have a question:

When on the "classified ViewportHistogram" we have the top 3 categories and others in gray. I don't see the others category showing in the histogram bars:

screen shot 2019-02-25 at 9 46 32 am

On the other hand, when "classified GlobalHistogram", the other category is showing in the bars:

screen shot 2019-02-25 at 9 47 42 am

Is this expected? Is it possible to add others to the viewport option as well?

thanks!
mamata

**Comment by elenatorro** [comment]
You're right, taking a look to it 👀

**Comment by elenatorro** [comment]
Since this PR has some important fixes, I think we should merge it and open a new one focused only on the examples for the histograms.

We planned to use the `vancouver_trees` dataset for both global and viewport examples, I'll add it in the new PR description.

File: src/renderer/viz/expressions/histogram/GlobalHistogram.js [hunk 1]
```diff
@@ -187,16 +251,13 @@ export default class GlobalHistogram extends Histogram {
...
-
-        histogram.forEach(feature => {
-            this._histogram.set(feature.value, feature.key);
+        const ratio = this._metadata.featureCount / this._metadata.sample.length;
+
+        this._metadata.sample.forEach((feature) => {
+            const key = feature[name];
+            const value = this._histogram.get(key) || 0;
+
+            this._histogram.set(key, value + ratio);
```
Comment (rjimenezda) [inline src/renderer/viz/expressions/histogram/GlobalHistogram.js]: math

File: src/renderer/viz/expressions/histogram/ViewportHistogram.js [hunk 2]
```diff
@@ -96,7 +98,7 @@ export default class ViewportHistogram extends Histogram {
      *
      * const s = carto.expressions;
      * const viz = new carto.Viz({
-     *   @histogram: s.viewportHistogram(s.prop('vehicles'))
+     *   @histogram: s.globalHistogram(s.prop('vehicles'))
```
Comment (rjimenezda) [inline src/renderer/viz/expressions/histogram/ViewportHistogram.js]:
```suggestion
     *   @histogram: s.viewportHistogram(s.prop('vehicles'))
```

File: src/renderer/viz/expressions/histogram/ViewportHistogram.js [hunk 3]
```diff
@@ -116,7 +118,7 @@ export default class ViewportHistogram extends Histogram {
      *
      * const s = carto.expressions;
      * const viz = new carto.Viz(`
-     *   @histogram: viewportHistogram($vehicles)
+     *   @histogram: globalHistogram($vehicles)
```
Comment (rjimenezda) [inline src/renderer/viz/expressions/histogram/ViewportHistogram.js]:
```suggestion
     *   @histogram: viewportHistogram($vehicles)
```

File: src/renderer/viz/expressions/histogram/ViewportHistogram.js [hunk 4]
```diff
@@ -130,12 +132,12 @@ export default class ViewportHistogram extends Histogram {
      * @example <caption>Get color values for the histogram when using a ramp.</caption>
      * const s = carto.expressions;
      * const viz = new carto.Viz(`
-     *   @histogram: s.viewportHistogram(s.prop('vehicles'))
+     *   @histogram: s.globalHistogram(s.prop('vehicles'))
```
Comment (rjimenezda) [inline src/renderer/viz/expressions/histogram/ViewportHistogram.js]:
```suggestion
     *   @histogram: s.viewportHistogram(s.prop('vehicles'))
```

File: src/renderer/viz/expressions/histogram/ViewportHistogram.js [hunk 5]
```diff
@@ -144,23 +146,93 @@ export default class ViewportHistogram extends Histogram {
      * // ]
      *
      * @example <caption>Get color values for the histogram when using a ramp. (String)</caption>
-     *
      * const s = carto.expressions;
      * const viz = new carto.Viz(`
-     *   @histogram: viewportHistogram($vehicles)
+     *   @histogram: globalHistogram($vehicles)
```
Comment (rjimenezda) [inline src/renderer/viz/expressions/histogram/ViewportHistogram.js]:
```suggestion
     *   @histogram: viewportHistogram($vehicles)
```

File: src/renderer/viz/expressions/histogram/ViewportHistogram.js [hunk 6]
```diff
@@ -144,23 +146,93 @@ export default class ViewportHistogram extends Histogram {
...
      * //   { frequency: 10, key: 'truck', value: { r: 95, g: 70, b: 144, a: 1 } }
      * //   { frequency: 20, key: 'bike', value: { r: 29, g: 105, b: 150, a: 1 } }
      * //   { frequency: 30, key: 'car', value: { r: 56, g: 166, b: 165, a: 1 } }
      * // ]
+     * @example <caption>Get color values for the histogram using a ramp with classified data.</caption>
+     * // Note: Both the ramp and the histogram expressions must use the same classification.
+     *
+     * const s = carto.expressions;
+     * const viz = new carto.Viz(`
+     *   @histogram: s.globalHistogram(s.top(s.prop('vehicles'), 2))
```
Comment (rjimenezda) [inline src/renderer/viz/expressions/histogram/ViewportHistogram.js]:
```suggestion
     *   @histogram: s.viewportHistogram(s.top(s.prop('vehicles'), 2))
```

File: src/renderer/viz/expressions/histogram/ViewportHistogram.js [hunk 7]
```diff
@@ -144,23 +146,93 @@ export default class ViewportHistogram extends Histogram {
...
+     * // [
+     * //   { frequency: 10, key: 'truck', value: { r: 95, g: 70, b: 144, a: 1 } }
+     * //   { frequency: 20, key: 'bike', value: { r: 29, g: 105, b: 150, a: 1 } }
+     * //   { frequency: 30, key: 'Others', value: { r: 0, g: 128, b: 0, a: 1 } }
+     * // ]
+     *
+     * @example <caption>Get color values for the histogram using a ramp with classified data (String).</caption>
+     * const s = carto.expressions;
+     * const viz = new carto.Viz(`
+     *   @histogram: globalHistogram(top($vehicles, 2))
```
Comment (rjimenezda) [inline src/renderer/viz/expressions/histogram/ViewportHistogram.js]:
```suggestion
     *   @histogram: viewportHistogram(top($vehicles, 2))
```

**Pull request merged**